packages/api/src/graphql/mutation.private.ts

```diff
@@ -1144,8 +1157,51 @@ export const GraphQLAdminMutation = new GraphQLObjectType<undefined, Context>({
           rejectionReason
         })
       }
+    },
+
+    // Settings
+    // ==========
+
+    updateSettingList: {
+      type: GraphQLList(GraphQLSetting),
+      args: {
+        value: {type: GraphQLList(GraphQLUpdateSettingArgs)}
+      },
+
+      async resolve(root, {value}, {authenticate, dbAdapter}) {
+        const {roles} = authenticate()
+        authorise(CanUpdateSettings, roles)
+
+        value.map(async ({name, value: val}: UpdateSettingArgs) => {
+          const fullSetting = await dbAdapter.setting.getSetting(name)
+
+          if (!fullSetting) {
+            throw new NotFound('setting', name)
+          }
+
+          const restrictions = fullSetting?.settingRestriction
+
+          const allowedValues = restrictions?.allowedValues
+            ? restrictions?.allowedValues?.includes(String(val))
+            : true
+          const inputLength =
+            restrictions?.inputLength !== undefined
```

Comment:
This wouldn't catch null/''/0

```suggestion
            restrictions?.inputLength
```